features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -358,6 +361,53 @@ internal class BitmapPoolTest {
         assertThat(captor.firstValue).isEqualTo(0)
     }
 
+    @RepeatedTest(3000)
+    fun `M not receive ConcurrentModificationException W get() { and onLowMemory }`() {
+        // when this occurs the frequency is roughly once per 3000 runs,
+
+        // Given
+        val threadList = mutableListOf<Thread>()
+        val numberOfThreads = 3
+        val allThreadsStartedLatch = CountDownLatch(numberOfThreads)
+        val allThreadsFinishedLatch = CountDownLatch(numberOfThreads)
+        testedCache.put(mockBitmap)
+        val bitmapKey = spyBitmapPoolHelper.generateKey(mockBitmap)
+        val failed = AtomicBoolean(false)
+
+        // When
+        repeat(numberOfThreads) {
+            if (Math.random() < 0.5) {
+                threadList.add(
+                Thread {
+                    allThreadsStartedLatch.countDown()
+                    allThreadsStartedLatch.await()
+                    try {
+                        testedCache.get(bitmapKey)
+                    } catch (e: ConcurrentModificationException) {
+                        failed.set(true)
```

Comment:
The disadvantage of this approach is that we are loosing the original exception and if test fails we have to somehow run it with debugger to be able to see the exception and its stack.

Slightly better approach, in my opinion at least, will be to use `Future` + `ExecutorService`. The you can do something like:

```
val executor = Executors.newFixedThreadPool(3)
val tasks = mutableListOf<Future<*>>()
repeat(...) {
   tasks += executor.submit { ... }
}

tasks.forEach { assertNotThrows { it.get() } }
```

This way you get original exception and a nice related message.

Reply:
Looks good - I'll make this change